From 48294abb840b4debcf8ffb4780d39edd06ee34b0 Mon Sep 17 00:00:00 2001
From: Mark Miller <mark@mirell.org>
Date: Thu, 16 Mar 2023 13:08:51 +0530
Subject: [PATCH 019/281] OpenWrt: 300-mips_expose_boot_raw.patch

mips: expose CONFIG_BOOT_RAW

This exposes the CONFIG_BOOT_RAW symbol in Kconfig. This is needed on
certain Broadcom chipsets running CFE in order to load the kernel.

Signed-off-by: Mark Miller <mark@mirell.org>
Acked-by: Rob Landley <rob@landley.net>

Change-Id: I3137654c137b726211f9e535acd0383403f4732a
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
---
 arch/mips/Kconfig | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index b26b77673c2c..7fdded9cf6db 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -1032,9 +1032,6 @@ config FW_ARC
 config ARCH_MAY_HAVE_PC_FDC
 	bool
 
-config BOOT_RAW
-	bool
-
 config CEVT_BCM1480
 	bool
 
@@ -3089,6 +3086,18 @@ choice
 		bool "Extend builtin kernel arguments with bootloader arguments"
 endchoice
 
+config BOOT_RAW
+	bool "Enable the kernel to be executed from the load address"
+	default n
+	help
+	 Allow the kernel to be executed from the load address for
+	 bootloaders which cannot read the ELF format. This places
+	 a jump to start_kernel at the load address.
+
+	 If unsure, say N.
+
+
+
 endmenu
 
 config LOCKDEP_SUPPORT
-- 
2.17.1

